### PR TITLE
Dynamic flatmap markers

### PR DESCRIPTION
**Pull request by Tehsurfer**
This can be viewed at https://algolia-maps.herokuapp.com/maps

### CODE REVIEW

File: src/mixins/ContentMixin.js [hunk 1]
```diff
@@ -60,13 +61,33 @@ export default {
...
+              // Keyword searches do not contain labels, so switch to keyword search if no label exists
+              returnedAction = {
+                type: "Search",
+                term: 'http://purl.obolibrary.org/obo/' + resource.feature.models.replace(':','_')
+              };
+              window.ra = returnedAction
+            } else {
+            // Facet search on anatomy if it is not a keyword search
+              returnedAction = {
+                type: "Facet",
```
Comment (alan-wu) [inline src/mixins/ContentMixin.js:84]: Can we change this so that we can use the Facets block as well?

File: src/mixins/ContentMixin.js [hunk 2]
```diff
@@ -60,13 +61,33 @@ export default {
...
+                facet: label,
```
Comment (alan-wu) [inline src/mixins/ContentMixin.js:85]: label is undefined here, I used to call this.idNamePair[resource.feature.models] to get the label instead.

File: src/components/viewers/MultiFlatmap.vue [hunk 3]
```diff
@@ -145,21 +149,92 @@ export default {
...
+        });
+      } else {
+        let terms = getAvailableTermsForSpecies(map.describes);
+        for (let i = 0; i < terms.length; i++) {
+          map.addMarker(terms[i].id, terms[i].type);
+        }
+      }
+    },
+    flatmapMarkerZoomUpdate(){
+      let flatmapImp = this.getFlatmapImp()
```
Comment (alan-wu) [inline src/components/viewers/MultiFlatmap.vue:196]: Calling the clearMarkers and addMarker at every zoom event can be intensive. Can we store the current zoom level and only update the markers when the zoom level has changed?

**Review by alan-wu — CHANGES_REQUESTED** [review]
I have commented on several places.
One minor thing is semi-colon. Please add them in for consistency please.

File: src/components/SplitFlow.vue [hunk 4]
```diff
@@ -104,27 +104,29 @@ export default {
...
           }
         } else if (action.type == "URL"){
           window.open(action.resource, '_blank')
-        } else if (action.type == "Neuron Search"){
-          this.$refs.sideBar.openNeuronSearch(action.resource)
-        } else if (action.type == "Facet") {        
-          const speciesFacets = [];
+        } else if (action.type == "Facet") {       
+          console.log('action', action) 
+          this.$refs.sideBar.addFilter(action);
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SplitFlow.vue:120]: the `Facet` action adds on facet to the current facets, as the `Facets` action clears all facets and sets new ones. I think we need boths

File: src/components/viewers/MultiFlatmap.vue (hunk 3 above)
Comment (Tehsurfer) [reply src/components/viewers/MultiFlatmap.vue:196]: yeah good idea

File: src/components/SplitFlow.vue [hunk 5]
```diff
@@ -104,27 +104,29 @@ export default {
...
+        } else if (action.type == "Facets") {
+          const facets = [];
           store.state.settings.facets.species.forEach(e => {
-            speciesFacets.push({facet: e, term:'species'});
+            facets.push({facet: e, term:'Species', facetPropPath: 'anatomy.organ.name'});
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SplitFlow.vue:124]: yeah you are right

File: src/mixins/ContentMixin.js (hunk 1 above)
Comment (Tehsurfer) [reply src/mixins/ContentMixin.js:84]: we want markers to add a filter to the current filter in the sidebar

Comment (Tehsurfer) [reply src/mixins/ContentMixin.js:84]: (which is what the 'Facet' action does. 'Facets' clears and sets a what array of facets is given)

File: src/components/viewers/MultiFlatmap.vue [hunk 6]
```diff
@@ -76,7 +82,13 @@ export default {
...
         type: this.entry.type,
       };
-      this.handleSyncPanZoomEvent(result)
+      let flatmapImp = this.getFlatmapImp();
+      let currentZoom = flatmapImp.getZoom()['zoom'];
+      if (this.zoomLevel !== currentZoom) {
+        this.zoomLevel = currentZoom;
+        this.flatmapMarkerZoomUpdate();
+      }
+      this.handleSyncPanZoomEvent(result);
```
Comment (alan-wu) [inline src/components/viewers/MultiFlatmap.vue:91]: Do not call handleSyncPanZoomEvent here please, that method should be called only when it is triggered by external callback.

**Review by alan-wu — APPROVED** [review]

**Pull request merged**